src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java

```diff
@@ -131,7 +132,15 @@ private void validate(final String offenderNo, final String accountCode, final L
         return entry -> (accountCode == null || entry.getAccountType().equals(accountCodeValue));
     }
 
-    public OffenderTransactionHistoryDto transform(final OffenderTransactionHistory offenderTransactionHistory) {
+    private OffenderTransactionHistory enrichRelatedTransactionsWithCurrentBalance(final OffenderTransactionHistory offenderTransactionHistory) {
+        var balanceAfterTransaction = new AtomicReference<>(offenderTransactionHistory.getCurrentBalance());
+        offenderTransactionHistory.getRelatedTransactionDetails().forEach(t ->
+            t.setCurrentBalance(balanceAfterTransaction.getAndUpdate(b -> b.subtract(t.getPayAmount())))
```

Comment:
I didn't order by date because the date is the same for all related TXs in a batch. Maybe we could order by sequence id - would that be better?

Reply:
Be worth checking to see what was done in prisonstaffhub

Reply:
It does order by date - but that just makes it harder to test, hence I chose to avoid that